### PR TITLE
Dispatch Workflow for AFQTS

### PR DESCRIPTION
**Pull request by shaheislamdfe**
### Context

https://trello.com/c/6KVsrKSx/2096-afqts-manual-deployment-workflow

Created addiitonal dispatch workflow to run deployments.

### Changes proposed in this pull request

Additional job which can deploy to any environment by providing the environment and docker tag.

### Guidance to review

Check workflow.
https://github.com/DFE-Digital/apply-for-qualified-teacher-status/actions/runs/11574489955/job/32218874448

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [x] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by shaheislamdfe** [comment]
[AFQTS: Manual deployment workflow](https://trello.com/c/6KVsrKSx/2096-afqts-manual-deployment-workflow)

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -8,11 +8,14 @@ on:
         required: true
         default: development
         type: environment
-        options:
-          - development
-          - test
-          - preproduction
-          - production
+      docker-image:
```
Comment (saliceti) [inline .github/workflows/deploy.yml:11]: Maybe rename to docker-image-tag for clarity?

**Comment by saliceti** [comment]
Tested here: https://github.com/DFE-Digital/apply-for-qualified-teacher-status/actions/runs/11591630307
Can we not run the build and test jobs?

**Review by saliceti — APPROVED** [review]
Tested successfully: https://github.com/DFE-Digital/apply-for-qualified-teacher-status/actions/runs/11598355449

**Pull request merged**